client-src/elements/chromedash-guide-editall-page.js

```diff
@@ -135,62 +132,64 @@ export class ChromedashGuideEditallPage extends LitElement {
     `;
   }
 
-  getStageFormFields(processStage) {
-    return FLAT_FORMS_BY_INTENT_TYPE[processStage.outgoing_stage] || [];
+  getStageForm(stageType) {
+    return FORMS_BY_STAGE_TYPE[stageType] || null;
   }
 
-  renderStageFormFields(formattedFeature, processStage, feStage, formFields) {
-    if (!formFields) return nothing;
-
-    return html`
-      <h3>${processStage.name}</h3>
-      <section class="flat_form">
-        ${formFields.map((field) => html`
-          <chromedash-form-field
-            name=${field}
-            stageId=${feStage.stage_id}
-            value=${formattedFeature[field]}>
-          </chromedash-form-field>
-        `)}
-      </section>
-    `;
+  // Flatten the sections to create a single array with all fields.
+  flattenSections(stageSections) {
+    return stageSections.reduce((combined, section) => [...combined, ...section.fields], []);
   }
 
-  getForms(formattedFeature, feStages) {
-    const formsToRender = [html`
-    <h3>Feature metadata</h3>
+  renderStageSection(formattedFeature, name, stageId, stageFields) {
+    if (!stageFields) return nothing;
+
+    return html`
+    <h3>${name}</h3>
     <section class="flat_form">
-      ${METADATA_FORM_FIELDS.map((field) => html`
+      ${stageFields.map(field => html`
         <chromedash-form-field
           name=${field}
+          stageId=${stageId}
           value=${formattedFeature[field]}>
         </chromedash-form-field>
       `)}
     </section>
-    `];
+    `;
+  }
+
+  getForms(formattedFeature, feStages) {
```

Comment:
This function is kind of hard to understand.  That may be an existing problem, but looking at it again now, it needs at least a docstring.

Reply:
Agreed, and a more thorough explanation has been added.